src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -2,20 +2,41 @@
 
 import org.junit.Before;
 import org.junit.Test;
+import java.math.*;
+import java.util.*;
+
 
 import static org.junit.Assert.assertTrue;
 
 public class ArrayListProductDaoTest
 {
     private ProductDao productDao;
-
+    
     @Before
     public void setup() {
         productDao = new ArrayListProductDao();
     }
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertTrue(!productDao.findProducts().isEmpty());
+    }
+    
+    @Test
+    public void testGetProduct(){
+        assertTrue(productDao.getProduct(4L)!=null);
+    }
+    
+    @Test
+    public void testSaveProduct(){
+        Product prod = new Product(14L, "nokia111", "Nokia 355", new BigDecimal(50), Currency.getInstance("USD"), 10, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg" );
+        productDao.save(prod);
+        assertTrue(productDao.getProduct(prod.getId())!=null);
+    }
+    
```

Comment:
Consider adding more tests on edge cases: for example, test to add two products with the same id.